lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -3177,6 +3202,25 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
           } else {
             FD->diagnose(diag::method_does_not_override, isClassContext)
                 .highlight(OA->getLocation());
+            //If the override is a no-args fun-decl, check if there exists
+            //a no-args var-decl in the base class, if yes, provide a fix-it.
+            if (FD->getParameters()->size() == 0) {
+              auto declName = FD->getBaseName();
+              auto declReturnType = FD->getResultInterfaceType()->getRValueType();
+              auto inherited = DC->getSelfClassDecl()->getInherited();
+              //If there is an inherited context, perform member lookup.
+              if (!inherited.empty()) {
+                auto superClass = inherited.front().getType();
+                constraints::ConstraintSystem CS(DC, None);
+                auto lookupResult = CS.performMemberLookup(swift::constraints::ConstraintKind::ValueMember, DeclNameRef(declName), superClass, FunctionRefKind::SingleApply, CS.getConstraintLocator({}), false);
```

Comment:
Same comment as about - the lookup should be done without constraint system involvement.